lib/archive_test.go

```diff
@@ -22,26 +22,30 @@ package lib
 
 import (
 	"bytes"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	null "gopkg.in/guregu/null.v3"
 )
 
-func TestAnonymizePath(t *testing.T) {
+func TestNormalizeAndAnonymizePath(t *testing.T) {
 	testdata := map[string]string{
-		"/tmp":                                      "/tmp",
-		"/tmp/":                                     "/tmp/",
-		"/tmp/myfile.txt":                           "/tmp/myfile.txt",
-		"/home/myname":                              "/home/nobody",
-		"/home/myname/":                             "/home/nobody/",
-		"/home/myname/myfile.txt":                   "/home/nobody/myfile.txt",
-		"/Users/myname/myfile.txt":                  "/Users/nobody/myfile.txt",
-		"/Documents and Settings/myname/myfile.txt": "/Documents and Settings/nobody/myfile.txt",
+		"/tmp":                                                "/tmp",
+		"/tmp/":                                               "/tmp/",
+		"/tmp/myfile.txt":                                     "/tmp/myfile.txt",
+		"/home/myname":                                        "/home/nobody",
+		"/home/myname/":                                       "/home/nobody/",
+		"/home/myname/foo/bar/myfile.txt":                     "/home/nobody/foo/bar/myfile.txt",
+		"/Users/myname/myfile.txt":                            "/Users/nobody/myfile.txt",
+		"/Documents and Settings/myname/myfile.txt":           "/Documents and Settings/nobody/myfile.txt",
+		"\\\\MYSHARED\\dir\\dir\\myfile.txt":                  "/nobody/dir/dir/myfile.txt",
+		"C:\\Users\\myname\\dir\\myfile.txt":                  "/C/Users/nobody/dir/myfile.txt",
+		"D:\\Documents and Settings\\myname\\dir\\myfile.txt": "/D/Documents and Settings/nobody/dir/myfile.txt",
 	}
 	for from, to := range testdata {
 		t.Run("path="+from, func(t *testing.T) {
-			assert.Equal(t, to, AnonymizePath(from))
+			assert.Equal(t, to, NormalizeAndAnonymizePath(from))
```

Comment:
`NormalizeAndAnonymizePath` should probably be idempotent, i.e. calling it on an already normalized path should not change it. I think we should add that to the test, something like:
```go
res := NormalizeAndAnonymizePath(from)
assert.Equal(t, to, res)
assert.Equal(t, res, NormalizeAndAnonymizePath(res))
```